File: 496. Next Greater Element I/496.py
class Solution(object):

    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = []
        for e in findNums:
            start = nums.index(e)
            find = False
            for i in range(start, len(nums)):
                if nums[i] > e:
                    ans += [nums[i]]
                    find = True
                    break
            if not find:
                ans += [-1]
        return ans

File: 496-Next Greater Element I/496.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& findNums, vector<int>& nums) {
        vector<int> ans;
        for(int i=0; i<findNums.size(); ++i){
            ptrdiff_t pos = find(nums.begin(), nums.end(), findNums[i]) - nums.begin();
            bool find = false;
            for(int j=pos; j<nums.size(); ++j){
                if(nums[j]>findNums[i]){
                    ans.push_back(nums[j]);
                    find = true;
                    break;
                }
            }
            if(!find) ans.push_back(-1);
        }
        return ans;
    }
};